lingua_franca/lang/parse_en.py

```diff
@@ -1685,3 +1710,256 @@ def numbers_to_digits(self, utterance):
+        .replace("a millennium", "1 millennium")
+
+    # we are always replacing 2 words, {N} {unit}
+    _replace_token = (replace_token + " " + replace_token) \
+        if replace_token else ""
+    # NOTE: above is a nasty hack, _convert_words_to_numbers normalized the
+    # string so we can do this, but this is essentially incorrect since each
+    # replaced number word should be replaced with a single replace_token
+
+    if time_unit == TimespanUnit.TIMEDELTA:
+        si_units = {
+            'microseconds': None,
+            'milliseconds': None,
+            'seconds': None,
+            'minutes': None,
+            'hours': None,
+            'days': None,
+            'weeks': None
+        }
+
+        units = ['months', 'years', 'decades', 'centurys', 'millenniums'] + \
+                list(si_units.keys())
+
+        for unit in units:
+            unit_pattern = pattern.format(
+                unit=unit[:-1])  # remove 's' from unit
+
+
+            matches = re.findall(unit_pattern, text)
+            value = sum(map(float, matches))
+            text = re.sub(unit_pattern, _replace_token, text)
+            if unit == "days":
+                if si_units["days"] is None:
+                    si_units["days"] = 0
+                si_units["days"] += value
+            elif unit == "months":
+                if si_units["days"] is None:
+                    si_units["days"] = 0
+                si_units["days"] += DAYS_IN_1_MONTH * value
+            elif unit == "years":
+                if si_units["days"] is None:
+                    si_units["days"] = 0
+                si_units["days"] += DAYS_IN_1_YEAR * value
+            elif unit == "decades":
+                if si_units["days"] is None:
+                    si_units["days"] = 0
+                si_units["days"] += 10 * DAYS_IN_1_YEAR * value
+            elif unit == "centurys":
+                if si_units["days"] is None:
+                    si_units["days"] = 0
+                si_units["days"] += 100 * DAYS_IN_1_YEAR * value
+            elif unit == "millenniums":
+                if si_units["days"] is None:
+                    si_units["days"] = 0
+                si_units["days"] += 1000 * DAYS_IN_1_YEAR * value
+            else:
+                si_units[unit] = value
+        duration = timedelta(**si_units) if any(si_units.values()) else None
+    elif time_unit in [TimespanUnit.RELATIVEDELTA,
+                       TimespanUnit.RELATIVEDELTA_APPROXIMATE,
+                       TimespanUnit.RELATIVEDELTA_FALLBACK,
+                       TimespanUnit.RELATIVEDELTA_STRICT]:
+        relative_units = {
+            'microseconds': None,
+            'seconds': None,
+            'minutes': None,
+            'hours': None,
+            'days': None,
+            'weeks': None,
+            'months': None,
+            'years': None
+        }
+
+        units = ['decades', 'centurys', 'millenniums', 'milliseconds'] + \
```

Comment:
plural of 'century' is 'centuries' and plural of 'millennium' is 'millennia'

Reply:
this is just an hack because of logic below stripping an s, to avoid the need for more code and reuse the existing loop